docs/sql-migration-guide.md

```diff
@@ -52,7 +52,7 @@ license: |
   
   - In Spark 3.1, the `schema_of_json` and `schema_of_csv` functions return the schema in the SQL format in which field names are quoted. In Spark 3.0, the function returns a catalog string without field quoting and in lower case. 
  
-  - In Spark 3.1, when `spark.sql.legacy.transformationPadNullWhenValueLessThenSchema` is true, Spark will pad NULL value when script transformation's output value size less then schema size in default-serde mode(script transformation with row format of `ROW FORMAT DELIMITED`). If false, Spark will keep original behavior to throw `ArrayIndexOutOfBoundsException`.
+  - In Spark 3.1, when script transformation output's value size less then schema size in default-serde mode(script transformation with row format of `ROW FORMAT DELIMITED`), Spark will pad NUll value to supplement data. In Spark 3.0 or earlier, Spark will do nothing and throw `ArrayIndexOutOfBoundsException`. To restore the behavior before Spark 3.1, you can set `spark.sql.legacy.transformationNotPadNullToSupplementData.enabled` to `true`.
```

Comment:
`value size less then` -> `value size is less than`

Reply:
```
In Spark 3.0 or earlier, Spark will do nothing and throw `ArrayIndexOutOfBoundsException`
```

I don't think we need a migration guide if we make a failure case working...

Reply:
And it's weird to provide a config to fail

Reply:
> `value size less then` -> `value size is less than`

Done

Reply:
> ```
> In Spark 3.0 or earlier, Spark will do nothing and throw `ArrayIndexOutOfBoundsException`
> ```
> 
> I don't think we need a migration guide if we make a failure case working...

This is a long standing error, and it's reasonable to point out that this is a problem that users might inadvertently trigger when they use Script Transform but don't know why. Inconsistent behavior in reversion upgrades can be confusing, and it's nice to point out specifically